HPXMLtoOpenStudio/tests/test_hvac_sizing.rb

```diff
@@ -621,6 +622,99 @@ def test_allow_increased_fixed_capacities
     end
   end
 
+  def test_autosizing_factors
+    clg_autosizing_factors = [0.8, 1.2]
+    htg_autosizing_factors = [0.9, 1.7]
+    for caf in clg_autosizing_factors
+      for haf in htg_autosizing_factors
+        args_hash = {}
+        args_hash['hpxml_path'] = File.absolute_path(@tmp_hpxml_path)
+
+        # Test air conditioner + furnace
+        hpxml, hpxml_bldg = _create_hpxml('base.xml')
+        hpxml_bldg.heating_systems[0].heating_capacity = nil
+        hpxml_bldg.cooling_systems[0].cooling_capacity = nil
+        XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
+        _model, _hpxml, hpxml_bldg = _test_measure(args_hash)
+        htg_cap_orig = hpxml_bldg.heating_systems[0].heating_capacity
+        clg_cap_orig = hpxml_bldg.cooling_systems[0].cooling_capacity
+        # apply autosizing factor
+        hpxml, hpxml_bldg = _create_hpxml('base.xml')
+        hpxml_bldg.heating_systems[0].heating_capacity = nil
+        hpxml_bldg.cooling_systems[0].cooling_capacity = nil
+        hpxml_bldg.heating_systems[0].heating_autosizing_factor = haf
+        hpxml_bldg.cooling_systems[0].cooling_autosizing_factor = caf
+        XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
+        _model, _hpxml, hpxml_bldg = _test_measure(args_hash)
+        htg_cap = hpxml_bldg.heating_systems[0].heating_capacity
+        clg_cap = hpxml_bldg.cooling_systems[0].cooling_capacity
+        assert_in_epsilon(hpxml_bldg.heating_systems[0].heating_capacity, htg_cap_orig * haf, 0.001)
+        assert_in_epsilon(hpxml_bldg.cooling_systems[0].cooling_capacity, clg_cap_orig * caf, 0.001)
+
+        # Test heat pump
+        hpxml, hpxml_bldg = _create_hpxml('base-hvac-air-to-air-heat-pump-1-speed.xml')
```

Comment:
Can you add one more test for a heat pump with a "separate" backup system?

Reply:
I assume the separate backup system should respond to its `HeatingAutosizingFactor` element and not the parent heat pump's `BackupHeatingAutosizingFactor`.

Reply:
I think maybe we should disallow specifying this in two places to avoid confusion, if it's a separate backup system, then disallow it in the parent object as `BackupHeatingAutosizingFactor` or disallow its child object to specify `HeatingAutosizingFactor`. I feel that it's may be easier to use `BackupHeatingAutosizingFactor` simply because the backup type is specified there and maybe easier to code the rule.

Reply:
> I feel that it's may be easier to use `BackupHeatingAutosizingFactor` simply because the backup type is specified there and maybe easier to code the rule.

But that is inconsistent with how all the other inputs (fuel type, efficiency, capacity, etc.) work. All the `BackupFoo` inputs only apply for integrated backup systems, and all the inputs in the `HeatingSystem` object apply for separate backup systems. But you raise a good point that we should probably require [here](https://github.com/NREL/OpenStudio-HPXML/blob/master/HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml#L1420) that none of the `BackupFoo` inputs are used when the backup type is separate.

Reply:
Sounds good, I'm going to add some rules there